odc/geo/gridspec.py

```diff
@@ -4,50 +4,74 @@
 # SPDX-License-Identifier: Apache-2.0
 """GridSpec class."""
 import math
-from typing import Iterator, Optional, Tuple
+from typing import Any, Dict, Iterator, Optional, Tuple, Union
 
 from affine import Affine
 
-from .crs import CRS
+from . import geom
+from .crs import SomeCRS, norm_crs_or_error
 from .geobox import GeoBox
 from .geom import BoundingBox, Geometry, intersects
+from .math import Bin1D
 
 
 class GridSpec:
     """
     Definition for a regular spatial grid.
 
-    :param CRS crs:
+    :param crs:
        Coordinate System used to define the grid
-    :param [float,float] tile_size:
-      ``(Y, X)`` size of each tile, in CRS units
-    :param [float,float] resolution:
-      ``(Y, X)`` size of each data point in the grid, in CRS units. ``Y`` will usually be negative.
-    :param [float,float] origin:
-      ``(Y, X)`` coordinates of a corner of the ``(0,0)`` tile in CRS units. default is ``(0.0, 0.0)``
+    :param tile_shape:
+       ``(Y, X)`` size of each tile in pixels
+    :param resolution:
+       ``(Y, X)`` size of each data point in the grid, in CRS units. ``Y`` will usually be negative
+    :param origin:
+       ``(X, Y)`` coordinate of a bottom-left corner of the ``(0,0)`` tile in CRS units. Default is
+       ``(0.0, 0.0)``
+    :param flipx: when ``True`` grid index for X axis increments left to right
+    :param flipy: when ``True`` grid index for Y axis increments top to bottom
     """
```

Comment:
I don't see it, but maybe I missed it -- is there a `to_transform()` method (or something like it) to get a [GDAL transform](https://gdal.org/tutorials/geotransforms_tut.html) from a `GridSpec`?

Reply:
There is none, because `GridSpec` is an "infinite" plane of tiles, there is a `transform` on each tile accessible via `GeoBox` class. So I guess you can say that `gs[0,0].geobox.transform` is a kind of "whole plane" transform, but not sure what flipping of axis does to that concept.

Think of grid spec more like an axis aligned 2-d binning abstraction, where each bin is an axis aligned geo registered bounded plane.